src/pages/theming/dark-mode.md

```diff
@@ -0,0 +1,254 @@
+---
+initialTab: 'preview'
+inlineHtmlPreviews: true
+previousText: 'Themes'
+previousUrl: '/docs/theming/themes'
+nextText: 'Advanced Theming'
+nextUrl: '/docs/theming/advanced'
+---
+
+# Dark Mode
+
+Ionic makes it easy to change the themes of your app, including supporting dark color schemes. With growing support for dark mode in native apps, developers are now looking to add it to their apps to support user preferences.
+
+## Using Media Queries
+
+The first way to enable dark mode is by using the CSS media query for the user's preferred color scheme. This media query will hook into the system setting of the user's device and apply the theme if a dark mode is enabled.
+
+```css
+@media (prefers-color-scheme: dark) {
+  :root {
+    /* dark mode variables go here */
+  }
+}
+```
+
+Currently, the `prefers-color-scheme` media query has [limited browser support](https://caniuse.com/#feat=prefers-color-scheme), so users will not be able to benefit from having the dark mode applied using this media query in certain browsers. However, the dark mode can still be applied by using a [class fallback](#class-fallback).
+
+
+## Class Fallback
+
+As a fallback method for devices that don't support the media query, the dark mode can be applied by styling a CSS selector and applying the class to the document body.
+
+```css
+@media (prefers-color-scheme: dark) {
+  :root {
+    /* dark mode variables go here */
+  }
+}
+
+/* Fallback for older browsers or manual mode */
+body.dark {
+  /* dark mode variables go here */
+}
+```
+
+With the variables targeting the `body.dark` selector, all that is needed now is to add the class to the `<body>` in the app. This can be done in a variety of ways depending on the framework your app is built with.
+
+Notice that the variables should be in both places in this example. We can [use JavaScript](#combining-with-javascript) in order to avoid setting the variables in two places.
+
+
+## Combining with JavaScript
+
+In order to keep the CSS variables written once and avoid having to update them in multiple places, the fallback and class can be combined by setting a variable in the media query and using it in the JavaScript to set the class on the document body based on the media query. Here's what the CSS would look like:
+
+```css
+@media (prefers-color-scheme: dark) {
+  :root {
+    --ion-color-scheme: dark;
+  }
+}
+
+body.dark {
+  /* dark mode variables go here */
+}
+```
+
+Notice that the variables above are only in the `body.dark` selector now, and the media query has one variable set. Now, in the JavaScript, the class can be added to the body based on the existence of this variable. This function should be called on load of an application, based on the framework being used:
+
+```javascript
+function onLoad() {
```

Comment:
this code block and the next are basically the same, minus moving code to the `toggleTheme` function. Normally I'm a fan of teaching/explaining more, but since they are so similar in this case, I suggest removing the "This could be improved upon by adding a toggle" line and just outright using one code block that has the toggleTheme func.

Reply:
Done :+1: